numpy/core/fromnumeric.py

```diff
@@ -1761,6 +1761,8 @@ def nonzero(a):
     """
     Return the indices of the elements that are non-zero.
 
+    NOTE - In place of using x[y.nonzero()], it is recommended to use x[y.astype(bool)].
+
```

Comment:
Please move this to after the `Return` section as
  
```
Notes
-----
To obtain the non-zero values, it is recommended to use ``x[y.astype(bool)]`` which
will correctly handle 0-d arrays.
```
and refactor the paragraph at 1766 to better express the idea that `nonzero` is the wrong tool for directly indexing the array.
If a line goes over 80 characters, break it up.

Reply:
@mattip Ping

Reply:
the text "The corresponding non-zero ..." on line 1767 above needs rephrasing or removal.